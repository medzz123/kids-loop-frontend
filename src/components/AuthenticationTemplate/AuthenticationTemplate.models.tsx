export interface AuthenticationTemplateProps {
  title: string;
}
