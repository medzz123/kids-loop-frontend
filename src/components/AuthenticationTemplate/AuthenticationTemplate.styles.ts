import { styled } from '@theme/config';

export const AuthenticationTemplateContainer = styled('div', {
  width: '$full',

  '@bp480': {
    m: 'auto',
    maxWidth: '$xs',
    minWidth: '$xs',
  },
});

export const ImageContainer = styled('div', {
  size: '$60',
  mb: '$40',
});

export const Elevation = styled('div', {
  backgroundColor: '$loContrast',
  boxShadow: '$colors$shadow',
  padding: '$20',
  borderRadius: '$sizes$12',
  display: 'flex',
  flexDirection: 'column',
  mb: '$12',
});

export const ActionsContainer = styled('div', {
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center',
  flexDirection: 'column',
  a: {
    marginLeft: '$12',
  },

  '@bp480': {
    flexDirection: 'row',
  },
});
