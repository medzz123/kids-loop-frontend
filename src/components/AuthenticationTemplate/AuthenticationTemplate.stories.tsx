import { Meta, Story } from '@storybook/react/types-6-0';
import React from 'react';

import AuthenticationTemplate from './AuthenticationTemplate';

export default {
  title: 'Components/AuthenticationTemplate',
  component: AuthenticationTemplate,
} as Meta;

const Template: Story = (storyArguments) => (
  <AuthenticationTemplate title="Story" {...storyArguments} />
);

export const Default = Template.bind({});
