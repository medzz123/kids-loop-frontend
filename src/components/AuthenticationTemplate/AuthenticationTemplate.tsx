import { Box } from '@components/Box';
import Link from '@components/Link';
import Menu from '@components/Menu';
import Text from '@components/Text';
import ThemeSwitch from '@components/ThemeSwitch';
import Image from 'next/image';
import { useRouter } from 'next/router';
import { useTranslation } from 'next-i18next';
import React, { FunctionComponent } from 'react';

import { AuthenticationTemplateProps } from './AuthenticationTemplate.models';
import {
  ActionsContainer,
  AuthenticationTemplateContainer,
  Elevation,
  ImageContainer,
} from './AuthenticationTemplate.styles';

const AuthenticationTemplate: FunctionComponent<AuthenticationTemplateProps> = (props) => {
  const { title, children } = props;

  const router = useRouter();

  const { t } = useTranslation('common', { useSuspense: false });

  const changeLocale = (locale: 'en' | 'de') => {
    router.push(router.asPath, router.asPath, { locale: locale });
  };

  return (
    <AuthenticationTemplateContainer>
      <Elevation data-testid="authenticationTemplate">
        <ImageContainer>
          <Image src="/logo.svg" width="60px" height="60px" />
        </ImageContainer>
        <Text variant="h1" level="h2" weight="bold" mb="large">
          {title}
        </Text>
        {children}
      </Elevation>
      <ActionsContainer>
        <Box css={{ display: 'flex', alignItems: 'center' }}>
          <Box css={{ mr: '$12' }}>
            <ThemeSwitch />
          </Box>
          <Menu
            label={t('languageSelect')}
            items={[
              {
                label: 'English',
                onClick: () => changeLocale('en'),
              },
              {
                label: 'Danish',
                onClick: () => changeLocale('de'),
              },
            ]}
          />
        </Box>
        <Box css={{ display: 'flex' }}>
          <Link href="/help" variant="secondary">
            {t('help')}
          </Link>
          <Link href="/privacy" variant="secondary">
            {t('privacy')}
          </Link>
          <Link href="/term" variant="secondary">
            {t('term')}
          </Link>
        </Box>
      </ActionsContainer>
    </AuthenticationTemplateContainer>
  );
};

export default AuthenticationTemplate;
