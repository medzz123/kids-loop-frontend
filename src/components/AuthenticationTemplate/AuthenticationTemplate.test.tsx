import { render } from '@test/library';
import React from 'react';

import AuthenticationTemplate from './AuthenticationTemplate';

test('AuthenticationTemplate Test', () => {
  const { container } = render(<AuthenticationTemplate title="Sign in" />);

  expect(container).toMatchSnapshot();
});
