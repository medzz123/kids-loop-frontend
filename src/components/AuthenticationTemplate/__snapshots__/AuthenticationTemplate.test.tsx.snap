// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenticationTemplate Test 1`] = `
<div>
  <div
    class="sxbp865"
  >
    <div
      class="sx8vaas"
      data-testid="authenticationTemplate"
    >
      <div
        class="sxvj4o5"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjAiIGhlaWdodD0iNjAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <noscript />
          <img
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
      </div>
      <h1
        class="sx3wn9e sx3wn9eaxhs1--level-h2 sx3wn9elue9b--weight-bold sx3wn9efxkwo--color-normal sx3wn9e0o7ux--align-left sx3wn9eyyv8n--spacing-large"
        data-testid="text"
      >
        Sign in
      </h1>
    </div>
    <div
      class="sxpzlvs"
    >
      <div
        class="sx03kze sx03kze-ju7t4"
      >
        <div
          class="sx03kze sx03kze-ordkl"
        >
          <button
            class="sxd7l8i"
            data-testid="themeSwitch"
            type="button"
          >
            <i
              class="sx03kze sx03kze-s82hm gg-sun"
            />
          </button>
        </div>
        <button
          aria-haspopup="menu"
          class="sxqt3zu"
          data-state="closed"
          type="button"
        >
          languageSelect
           
          <i
            class="sx03kze sx03kze-8rfvp gg-chevron-down"
          />
        </button>
      </div>
      <div
        class="sx03kze sx03kze-pagfp"
      />
      <div
        class="sx03kze sx03kze-2cs53"
      >
        <a
          class="sx28ds4 sx28ds4fxkwo--type-secondary"
          data-testid="link"
        >
          help
        </a>
        <a
          class="sx28ds4 sx28ds4fxkwo--type-secondary"
          data-testid="link"
        >
          privacy
        </a>
        <a
          class="sx28ds4 sx28ds4fxkwo--type-secondary"
          data-testid="link"
        >
          term
        </a>
      </div>
    </div>
  </div>
</div>
`;
